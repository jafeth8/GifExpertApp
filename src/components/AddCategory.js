import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setcategories}) => {
    const [inputValue, setinputValue] = useState('')

    const handleInputChange=(e)=> {
        //console.log(e.target.value);
        setinputValue(e.target.value);
        //console.log("handleInputChange() ejecutado")//temporal para pruebas
    }

    const handleSubmit=(e)=> {
        e.preventDefault();
        console.log('handle event ejecutado');
        
        if(inputValue.trim().length>2){
            setcategories(cats=>[inputValue,...cats]);
            setinputValue('');
        }
        
    }
    return (
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />  
        </form>
    )
}

AddCategory.propTypes = {
    setcategories:PropTypes.func.isRequired,
}
