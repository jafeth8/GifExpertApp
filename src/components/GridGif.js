import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'

import { GifGridItem } from './GifGridItem';

export const GridGif = ({category}) => {

    
    const {data:images,loading}=useFetchGifs(category)
    console.log("cargando componente");
     
    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>
            {loading?<p className="animate__animated animate__flash" >loading..</p>:null}
            <div className="card-grid">
            
                {
                    images.map(img =>(
                        <GifGridItem 
                        key={img.id} 
                        //img={img}
                        {...img}
                        />
                    ))
                }
            
            </div>

        </>
        
    )
}
