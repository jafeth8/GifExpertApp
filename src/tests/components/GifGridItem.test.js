import "@testing-library/jest-dom"
import {shallow} from "enzyme"
import { GifGridItem } from "../../components/GifGridItem"

describe('pruebas en GifGridItem.js', () => {
    const titulo="jibril";
    const url="http://localhost/jibril.jpg";

    test('mostrar correctamente el componente', async () => {
        const wrapper = shallow(<GifGridItem title={titulo} url={url}/>)
        expect(wrapper).toMatchSnapshot()
    })
    
})

