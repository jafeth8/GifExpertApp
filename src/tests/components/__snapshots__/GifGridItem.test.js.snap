// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifGridItem.js mostrar correctamente el componente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="jibril"
    src="http://localhost/jibril.jpg"
  />
  <p>
    jibril
  </p>
</div>
`;
